src/material/core/focus_indicator/_focus-indicator.scss

```diff
@@ -0,0 +1,61 @@
+@import '../theming/theming';
+
+@mixin mat-strong-focus-indicators() {
+  // Base styles for focus indicators.
+  .mat-focus-indicator::before {
+    border-radius: 4px;
+    border-width: 2px;
+    border-style: solid;
+    box-sizing: border-box;
+    position: absolute;
+    pointer-events: none;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+
+  // Default insets/outsets for focus indicators.
+  .mat-focus-indicator {
+    &.mat-button-base::before,
+    &.mat-card::before,
+    &.mat-sort-header-button::before {
+      margin: -4px;
+    }
+
+    &.mat-calendar-body-cell::before {
+      margin: -2px;
+    }
+
+    &.mat-tab-link::before,
+    &.mat-tab-label::before {
+      margin: 4px;
+    }
+  }
+
+  // On focus, actually render the focus indicators. Defining a pseudo
+  // element's content will cause it to render.
+  .mat-focus-indicator {
+    // Checkbox, Option, Slide Toggle
+    .cdk-focused.mat-checkbox &::before,
+    .cdk-focused.mat-radio-button &::before,
+    .cdk-focused.mat-slide-toggle &::before,
+
+    // Option
+    &.mat-option.mat-active::before,
+
+    // All other components
+    &:focus::before {
+      content: '';
+    }
+  }
+}
+
+@mixin mat-strong-focus-indicators-theme($themeOrColor) {
+  .mat-focus-indicator::before {
```

Comment:
I would add a comment here that explains the idea of accepting a theme or a concrete color